osbs/tekton.py

```diff
@@ -699,6 +699,14 @@ def wait_for_taskruns(self):
                 if task_run_name not in watched_task_runs:
                     watched_task_runs.add(task_run_name)
                     yield task_run_data['pipelineTaskName'], task_run_name
+            try:
+                status = pipeline_run['status']['conditions'][0]['status']
+            except KeyError:
+                logger.error("pipeline run does not have any status")
+                return
```

Comment:
Should we `return` here? The other method (`wait_for_start`) keeps waiting if there's no status

Reply:
wait_for_start is waiting for prun to start, and AT this point in wait_for_tasks, prun already started so it has status, and if not something broke with prun I'd say

Reply:
Hmm, maybe raise an error then? Or don't even catch the KeyError and let it fail?

Reply:
Hmm, alright then. I don't like ignoring this kind of error (one that should never happen), but fair enough :)